src/test/groovy/org/prebid/server/functional/service/PrebidServerService.groovy

```diff
@@ -352,21 +353,19 @@ class PrebidServerService implements ObjectMapperWrapper {
         }
     }
 
-    List<String> getLogsByTime(Instant testStart,
-                               Instant testEnd = Instant.now()) {
-        if (!testEnd.isAfter(testStart)) {
+    List<String> getLogsByTime(Instant testStart, Instant testEnd = Instant.now()) {
+        if (testEnd.isBefore(testStart)) {
             throw new IllegalArgumentException("The end time of the test is less than the start time")
         }
-
-        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss")
-                                                       .withZone(ZoneId.from(UTC))
+        def formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss")
+                                         .withZone(ZoneId.from(UTC))
         def logs = Arrays.asList(pbsContainer.logs.split("\n"))
         def filteredLogs = []
 
-        def deltaTime = Duration.between(testStart, testEnd).seconds
+        def deltaTime = Duration.between(testStart, testEnd).toMillis()
 
-        for (int i = 0; i <= deltaTime; i++) {
-            def time = testStart.plusSeconds(i)
+        for (int i = 0; i <= deltaTime; i += deltaTime / 5) {
```

Comment:
I don't like approach with millis. On large ranges you will be doing a lot of pointless iterations. You can just do `deltaTime + 1` as a stop condition. That way you will make sure that you are properly handling tails. Because this is not an array, you don't need to worry about going out of bounds.